pxtpy/converter.ts

```diff
@@ -2305,21 +2305,22 @@ namespace pxt.py {
     }
 
     export interface Py2TsRes {
-        generated: Map<string>,
-        diagnostics: pxtc.KsDiagnostic[]
+        diagnostics: pxtc.KsDiagnostic[],
+        success: boolean,
+        outfiles: { [key: string]: string }
     }
     export function py2ts(opts: pxtc.CompileOptions): Py2TsRes {
         let modules: py.Module[] = []
-        const generated: Map<string> = {}
+        const outfiles: Map<string> = {}
         diagnostics = []
 
         if (!opts.sourceFiles)
-            error(null, 9566, lf("missing sourceFiles"));
+            throw Error("missing sourceFiles! Cannot convert py to ts")
```

Comment:
no more error number?

Reply:
In this case it wasn't helpful since this isn't anything to do with the user's code, it's an internal error that should never happen. I was the one that added the error number above but really it should be fatal.

Reply:
actually if we don't throw here, the missing property causes a null deref exception before the error is returned. That's why I converted this to a throw because there's no chance to get the error information out anyway, this just catches the error earlier.